web/src/filter.ts

```diff
@@ -1564,6 +1560,16 @@ export class Filter {
         // We could turn it into something more like a compiler:
         // build JavaScript code in a string and then eval() it.
 
+        // Make a shallow copy to avoid modifying the original terms.
+        const terms = this._terms.map((term) => ({...term}));
+        // Since we are doing a case-insensitive match, we need to
+        // ensure that the operand is in lowercase.
+        for (const term of terms) {
+            if (term.operand !== undefined) {
+                term.operand = term.operand.toLowerCase();
+            }
+        }
```

Comment:
So I don't think we have a guarantee that every term type is case-insensitive; we probably should only do this for those terms where we have a reason to believe it is, like the topic/channel terms.